packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -488,9 +488,16 @@ export const makeZCFZygote = async (
           publicFacet = undefined,
           creatorInvitation = undefined,
         }) => {
-          const priorCreatorFacet = zcfBaggage.get('creatorFacet');
-          const priorPublicFacet = zcfBaggage.get('publicFacet');
-          const priorCreatorInvitation = zcfBaggage.get('creatorInvitation');
+          let priorCreatorFacet;
+          let priorPublicFacet;
+          let priorCreatorInvitation;
+          try {
+            priorCreatorFacet = zcfBaggage.get('creatorFacet');
+            priorPublicFacet = zcfBaggage.get('publicFacet');
+            priorCreatorInvitation = zcfBaggage.get('creatorInvitation');
+          } catch (e) {
+            Fail`restartContract failed: original contract facets were not durable`;
+          }
```

Comment:
Is it really the case that the only exception possible is that the facets were not durable? I suppose so... `zcfBaggage.get(...)` is the only thing in the `try`.

But if `get(...)` had a bug, we might lose diagnostic info here. It feels weird to throw away `e` completely.
log it?

Reply:
Do we support something like this?

```suggestion
          } catch (e) {
            throw assert.error(`restartContract failed: original contract facets were not durable`, { cause: e });
          }
```

Reply:
In what way would that be better?

I think `Fail` gets us an error, a stack, and messages visible in logs. I don't know what `throw assert.error()` could add to that. Whatever `assert.error()` does, I wouldn't expect it to return something that would be useful to `throw`, but I certainly don't know for sure.

Reply:
> In what way would that be better?

It doesn't throw away the details of `e`. Note especially `{ cause: e }`